packages/paste-core/components/checkbox/src/Checkbox.tsx

```diff
@@ -90,15 +100,29 @@ const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
     const disabled = props.disabled != null ? props.disabled : checkboxGroupContext.disabled;
     const name = props.name != null ? props.name : checkboxGroupContext.name;
     const hasError = props.hasError != null ? props.hasError : checkboxGroupContext.hasError;
-    const handleChange = props.onChange != null ? props.onChange : checkboxGroupContext.onChange;
+
+    // Manages what the value of checked is depending on if it is controlled or uncontrolled
+    const [checkedState, setCheckedState] = React.useState(!!defaultChecked);
```

Comment:
This is a simplified version of what Chakra does (see [their useCheckbox hook](https://github.com/chakra-ui/chakra-ui/blob/e1057b085f47916fb148561e40fe0b4afbdd633f/packages/checkbox/src/use-checkbox.ts#L160))

If the user passes checked & onChange it is a controlled component, if they don't pass them it is uncontrolled.

Reply:
because `defaultChecked` is a boolean, what's the difference between using the double bang operator and just saying `useState(defaultChecked)`? wouldn't that still give you either `true` or `false`?